src/components/PostDetails.tsx

```diff
@@ -158,7 +104,7 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
         )}
 
         {loaded && !hasError && visible && (
-          <NewCommentForm onSubmit={addComment} />
+          <NewCommentForm onSubmit={onSubmit} />
```

Comment:
The `NewCommentForm` component is rendered when the `visible` state is true. This is a good way to conditionally render components, but consider adding an 'Cancel' button or a similar user interface element to allow the user to hide the form without submitting a comment.